messagewatch/messagewatch.py

```diff
@@ -0,0 +1,196 @@
+from datetime import datetime, timedelta, timezone
+from typing import Optional, List
+
+import discord
+from redbot.core import Config, checks
+from redbot.core.bot import Red
+from redbot.core import commands
+from redbot.core.utils.mod import is_mod_or_superior
+
+
+class MessageWatchCog(commands.Cog):
+    """MessageWatch Cog"""
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=128986274420752384004)
+        self.embed_speeds = {}
+        default_guild_config = {
+            "logchannel": "",  # Channel to send alerts to
+            "recent_fetch_time": 15000,  # Time in milliseconds to fetch recent prior embed times used for calculations.
+            "frequencies": {  # Collection of allowable frequencies
+                "embed": 1  # Allowable frequency for embeds
+            },
+            "exemptions": {
+                "member_duration": 30,  # Minimum member joined duration required to qualify for any exemptions
+                "text_messages": 1,  # Minimum text-only message frequency required to exempt a user
+            }
+        }
+
+        self.config.register_guild(**default_guild_config)
+
+    @checks.admin()
+    @commands.group("messagewatch", aliases=["mw"], pass_context=True)
+    async def _messagewatch(self, ctx: commands.Context):
+        pass
+
+    @_messagewatch.command(name="logchannel")
+    async def _messagewatch_logchannel(self, ctx: commands.Context, channel: Optional[discord.TextChannel]):
+        """Set/update the channel to send message activity alerts to."""
+
+        chanId = ctx.channel.id
+        if channel:
+            chanId = channel.id
+        await self.config.guild(ctx.guild).logchannel.set(chanId)
+        await ctx.send("✅ Alert channel successfully updated!")
+
+    @_messagewatch.command(name="fetchtime")
+    async def _messagewatch_fetchtime(self, ctx: commands.Context, time: str):
+        """Set/update the recent message fetch time (in milliseconds)."""
+        try:
+            val = float(time)
+            await self.config.guild(ctx.guild).recent_fetch_time.set(val)
+            await ctx.send("Recent message fetch time successfully updated!")
+        except ValueError:
+            await ctx.send("Recent message fetch time FAILED to update. Please specify a `float` value only!")
+
+    @_messagewatch.group("frequencies", aliases=["freq", "freqs"], pass_context=True)
+    async def _messagewatch_frequencies(self, ctx: commands.Context):
+        pass
+
+    @_messagewatch_frequencies.command(name="embed")
+    async def _messagewatch_frequencies_embed(self, ctx: commands.Context, frequency: str):
+        """Set/update the allowable embed frequency."""
+        try:
```

Comment:
I guess this whole thing is because you want to return a better error message than the default `float` converter that discord.py provides... I think this would better be handled as a custom converter. Below code might not be accurate, but something like this should do.
```py
def custom_float_converter(message: str):
    async def convert(ctx: commands.Context, arg: str) -> float:
        try:
            return float(arg)
        except ValueError:
            raise CommandError(message)

    return convert

async def command(self, ctx: commands.Context, value: custom_float_converter("Recent message fetch time FAILED to update. Please specify a `float` value only!")):
    ...
```
The best part about it is that this logic is reusable, which should reduce duplicate code a fair bit.